engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/util/TimeCharacteristicCompatibility.scala

```diff
@@ -0,0 +1,27 @@
+package pl.touk.nussknacker.engine.flink.util
+
+import com.github.ghik.silencer.silent
+import org.apache.flink.streaming.api.TimeCharacteristic
+import org.apache.flink.streaming.api.scala.StreamExecutionEnvironment
+import pl.touk.nussknacker.engine.flink.api.timestampwatermark.TimestampWatermarkHandler
+
+import scala.annotation.nowarn
+
+/*
+Setting time characteristic is not needed and deprecated in Flink >= 1.12. We use this class not to pollute codebase
+  with deprecation warnings, while remaining compatible with older Flink versions. If Flink removes it completely (which probably
+  won't happen soon), we'll introduce other mechanism to handle it. In its current form it's not well defined for joins anyway
+ */
+// Remove @silent after upgrade to silencer 1.7
+@silent("deprecated")
+@nowarn("deprecated")
+object TimeCharacteristicCompatibility {
```

Comment:
Shouldn't we have just an interface and empty implementation here and this implementation in nk-flink-compat? Thanks to that, no silencing would be needed - maybe only `@deprecated` annotation. Also WDYT about adding a note, that this class will be removed in further versions? It will be necessary when we will add deprecation

Reply:
I removed this class altogether :) Thanks for pointing out that it's kind of smelly :+1: 
I moved the logic to compatibility layer...